src/Api/Vault/AuthorizationHandlers/Collections/CollectionAuthorizationHandler.cs

```diff
@@ -0,0 +1,185 @@
+﻿using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.Utilities;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Bit.Api.Vault.AuthorizationHandlers.Collections;
+
+public class CollectionAuthorizationHandler : BulkAuthorizationHandler<CollectionOperationRequirement, Collection>
+{
+    private readonly ICurrentContext _currentContext;
+    private readonly ICollectionRepository _collectionRepository;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+
+    public CollectionAuthorizationHandler(ICurrentContext currentContext, ICollectionRepository collectionRepository, IOrganizationUserRepository organizationUserRepository)
+    {
+        _currentContext = currentContext;
+        _collectionRepository = collectionRepository;
+        _organizationUserRepository = organizationUserRepository;
+    }
+
+    protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        CollectionOperationRequirement requirement, ICollection<Collection> resources)
+    {
+        switch (requirement)
+        {
+            case not null when requirement == CollectionOperations.Create:
+                await CanCreateAsync(context, requirement, resources);
+                break;
+
+            case not null when requirement == CollectionOperations.Delete:
+                await CanDeleteAsync(context, requirement, resources);
+                break;
+
+            case not null when requirement == CollectionOperations.ModifyAccess:
+                await CanManageCollectionAccessAsync(context, requirement, resources);
+                break;
+        }
+    }
+
+    private async Task CanCreateAsync(AuthorizationHandlerContext context, CollectionOperationRequirement requirement, ICollection<Collection> resources)
+    {
+        // Bulk creation of collections is not supported
+        if (resources.Count > 1)
```

Comment:
thought: Not sure if this is relevant to the authorization logic and should instead be handled by command/service business logic. I don't see any particular reason we couldn't authorize the creation of multiple collections at once? Just need to check that the all belong to the same organization like the other operations check.

I imagine it could be useful in the future, so we can use this auth handler for creating multiple collections during a data import or something similar.

Reply:
I agree that it takes minimal effort to make this method handle bulk resources. I've updated the method accordingly.